site/pages/retirement/[beneficiary_address]/[retirement_index].tsx

```diff
@@ -0,0 +1,64 @@
+import { GetStaticProps } from "next";
+
+import { queryKlimaRetireByIndex } from "@klimadao/lib/utils";
+import { SingleRetirementPage } from "components/pages/Retirement/SingleRetirement";
+import { loadTranslation } from "lib/i18n";
+import { IS_PRODUCTION } from "lib/constants";
+
+export const getStaticProps: GetStaticProps = async (ctx) => {
+  try {
+    if (IS_PRODUCTION) {
+      throw new Error("Not on Staging");
+    }
+
+    const { params, locale } = ctx;
+
+    if (
+      !params ||
+      (!params?.beneficiary_address && !params?.retirement_index) ||
+      (!!params.retirement_index && Number(params.retirement_index) === NaN)
+    ) {
+      throw new Error("No params found");
+    }
+
+    const retirementIndex = Number(params.retirement_index) - 1; // totals does not include index 0
+
+    const retirement = await queryKlimaRetireByIndex(
+      params?.beneficiary_address as string,
+      retirementIndex
+    );
+
+    if (!retirement) {
+      throw new Error("No retirement found");
+    }
+
+    const translation = await loadTranslation(locale);
```

Comment:
we can run this and the query above in parallel for minor performance boost with `Promise.all()`

Reply:
Done